docs/source/app-dev/command-deduplication.rst

```diff
@@ -0,0 +1,364 @@
+
+- The :ref:`submission ID <com.daml.ledger.api.v1.Commands.submission_id>` is chosen by the application to identify a specific submission.
+  It is included in the corresponding completion event so that the application can correlate specific submissions to specific completions.
+  An application should never reuse a submission ID.
+
+The ledger may arbitrarily extend the deduplication period specified in the submission, even beyond the maximum deduplication time specified in the :ref:`ledger configuration <ledger-configuration-service>`.
+
+.. note::
+   The maximum deduplication time is the length of the deduplication period guaranteed to be supported by the participant.
+   
+The deduplication period chosen by the ledger is the *effective deduplication period*.
+The ledger may also convert a requested deduplication duration into an effective deduplication offset or vice versa.
+The effective deduplication period is reported in the command completion event in the :ref:`deduplication duration <com.daml.ledger.api.v1.Completion.deduplication_duration>` or :ref:`deduplication offset <com.daml.ledger.api.v1.Completion.deduplication_offset>` fields.
+
+A command submission is considered a **duplicate submission** if at least one of the following holds:
+
+- The submitting participant's completion service contains a successful completion event for the same :ref:`change ID <change-id>` within the *effective* deduplication period.
+
+- The participant or Daml ledger are aware of another command submission in-flight with the same :ref:`change ID <change-id>` when they perform command deduplication.
+
+The outcome of command deduplication is communicated as follows:
+
+- Command submissions via the :ref:`command service <command-service>` indicate the command deduplication outcome as a synchronous gRPC response unless when the `gRPC deadline <https://grpc.io/blog/deadlines/>`_ was exceeded.
+
+  .. note::
+     The outcome MAY additionally appear as a completion event on the :ref:`command completion service <command-completion-service>`,
+     but applications using the :ref:`command service <command-service>` typically need not process completion events.
+
+- Command submissions via the :ref:`command submission service <command-submission-service>` can indicate the outcome as a synchronous gRPC response,
+  or asynchronously through the :ref:`command completion service <command-completion-service>`.
+  In particular, the submission may be a duplicate even if the command submission service acknowledges the submission with the gRPC status code ``OK``.
+
+Independently of how the outcome is communicated, command deduplication generates the following outcomes of a command submission:
+
+- If there is no conflicting submission with the same :ref:`change ID <change-id>` on the Daml ledger or in-flight, the completion event and possibly the response convey the result of the submission (success or a gRPC error).
```

Comment:
Shall we link the page for [gRPC status codes](https://grpc.github.io/grpc/core/md_doc_statuscodes.html) here? I.e., it may be helpful to have the context around what the gRPC status codes below mean.

Reply:
I'll link to our own page about error codes, which also explains the gRPC status codes and the Daml error codes, along with a link to the official page.